packages/@tinacms/datalayer/src/gitProviders.ts

```diff
@@ -0,0 +1,84 @@
+import { Octokit } from '@octokit/rest'
+import { Base64 } from 'js-base64'
+
+export interface GitProvider {
+  onPut: (key: string, value: string) => Promise<void>
+  onDelete: (key: string) => Promise<void>
+}
+
+export interface GitProviderOptions {
+  owner: string
+  repo: string
+  token: string
+  branch: string
+}
+
+export class GitHubProvider implements GitProvider {
+  octokit: Octokit
+  owner: string
+  repo: string
+  branch: string
+
+  constructor(args: GitProviderOptions) {
+    this.owner = args.owner
+    this.repo = args.repo
+    this.branch = args.branch
+    this.octokit = new Octokit({
+      auth: args.token,
+    })
+  }
+
+  async onPut(key: string, value: string) {
```

Comment:
The getContent calls here need to use `ref` and not `branch`. See https://github.com/tinacms/tina-self-hosted-demo/blob/main/tina/database.ts#L39